### PR TITLE
Erc20 Unit Tests

### PR DESCRIPTION
**Pull request by kantagara**

### CODE REVIEW

**Review by juans-chainsafe — COMMENTED** [review]
I dont get those errors anymore, and tests are working locally!
Screenshot 2023-09-11 at 13 04 05

But I get all these errors if I try to build with WebGL:
```
Assets/Tests/Erc20Tests.cs(16,6): error CS0246: The type or namespace name 'UnitySetUpAttribute' could not be found (are you missing a using directive or an assembly reference?)

Assets/Tests/Erc20Tests.cs(16,6): error CS0246: The type or namespace name 'UnitySetUp' could not be found (are you missing a using directive or an assembly reference?)

Assets/Tests/Erc20Tests.cs(36,6): error CS0246: The type or namespace name 'UnityTestAttribute' could not be found (are you missing a using directive or an assembly reference?)

Assets/Tests/Erc20Tests.cs(36,6): error CS0246: The type or namespace name 'UnityTest' could not be found (are you missing a using directive or an assembly reference?)

Assets/Tests/Erc20Tests.cs(48,6): error CS0246: The type or namespace name 'UnityTestAttribute' could not be found (are you missing a using directive or an assembly reference?)

Assets/Tests/Erc20Tests.cs(48,6): error CS0246: The type or namespace name 'UnityTest' could not be found (are you missing a using directive or an assembly reference?)

Assets/Tests/Erc20Tests.cs(57,6): error CS0246: The type or namespace name 'UnityTestAttribute' could not be found (are you missing a using directive or an assembly reference?)

Assets/Tests/Erc20Tests.cs(57,6): error CS0246: The type or namespace name 'UnityTest' could not be found (are you missing a using directive or an assembly reference?)

Assets/Tests/Erc20Tests.cs(66,6): error CS0246: The type or namespace name 'UnityTestAttribute' could not be found (are you missing a using directive or an assembly reference?)

Assets/Tests/Erc20Tests.cs(66,6): error CS0246: The type or namespace name 'UnityTest' could not be found (are you missing a using directive or an assembly reference?)

Assets/Tests/Erc20Tests.cs(80,6): error CS0246: The type or namespace name 'UnityTestAttribute' could not be found (are you missing a using directive or an assembly reference?)

Assets/Tests/Erc20Tests.cs(80,6): error CS0246: The type or namespace name 'UnityTest' could not be found (are you missing a using directive or an assembly reference?)

Assets/Tests/Erc20Tests.cs(90,6): error CS0246: The type or namespace name 'UnityTestAttribute' could not be found (are you missing a using directive or an assembly reference?)

Assets/Tests/Erc20Tests.cs(90,6): error CS0246: The type or namespace name 'UnityTest' could not be found (are you missing a using directive or an assembly reference?)

Error building Player because scripts had compiler errors

Build completed with a result of 'Failed' in 5 seconds (5247 ms)
UnityEngine.GUIUtility:ProcessEvent (int,intptr,bool&) (at /Users/bokken/build/output/unity/unity/Modules/IMGUI/GUIUtility.cs:189)

UnityEditor.BuildPlayerWindow+BuildMethodException: 15 errors
  at UnityEditor.BuildPlayerWindow+DefaultBuildMethods.BuildPlayer (UnityEditor.BuildPlayerOptions options) [0x002ce] in /Users/bokken/build/output/unity/unity/Editor/Mono/BuildPlayerWindowBuildMethods.cs:193 
  at UnityEditor.BuildPlayerWindow.CallBuildMethods (System.Boolean askForBuildLocation, UnityEditor.BuildOptions defaultBuildOptions) [0x00080] in /Users/bokken/build/output/unity/unity/Editor/Mono/BuildPlayerWindowBuildMethods.cs:94 
UnityEngine.GUIUtility:ProcessEvent (int,intptr,bool&) (at /Users/bokken/build/output/unity/unity/Modules/IMGUI/GUIUtility.cs:189)
```
@kantagara

**Review by juans-chainsafe — DISMISSED** [review]
Errors fixed

**Comment by sneakzttv** [comment]
linter is failing, whitespaces :D

**Review by sneakzttv — APPROVED** [review]

File: src/ChainSafe.GamingSdk.EVM.Unity/publish-to-unity-package.sh
Comment (robGG1997) [inline src/ChainSafe.GamingSdk.EVM.Unity/publish-to-unity-package.sh:1]: is this change necessary? `setup.sh` uses this bash script we shouldn't change it unless we need to

File: src/UnitySampleProject/Assets/Samples/web3.unity SDK/2.5.0-pre001/Web3.Unity Samples/Scripts/Editor/Tests.meta
Comment (robGG1997) [inline src/UnitySampleProject/Assets/Samples/web3.unity SDK/2.5.0-pre001/Web3.Unity Samples/Scripts/Editor/Tests.meta:1]: could we update the hierarchy/directory for tests based on [this](https://docs.unity3d.com/Manual/cus-tests.html), which should fix the namespace build errors, you can also merge to #603 and work from there

**Review by robGG1997 — CHANGES_REQUESTED** [review]

File: src/UnitySampleProject/Assets/Samples/web3.unity SDK/2.5.0-pre001/Web3.Unity Samples/Scripts/Editor/Tests.meta
Comment (kalambet) [reply src/UnitySampleProject/Assets/Samples/web3.unity SDK/2.5.0-pre001/Web3.Unity Samples/Scripts/Editor/Tests.meta:1]: Yes, 603 should be merged first, I agree.

File: src/ChainSafe.GamingSdk.EVM.Unity/publish-to-unity-package.sh
Comment (kalambet) [reply src/ChainSafe.GamingSdk.EVM.Unity/publish-to-unity-package.sh:1]: Yes, running `setup.sh` every time include unnecessary git submodules initialization. So after all the source changes in order to re-publish built sources I'm using both `publish-to-unity-package.sh` and `debug-publish-to-unity-package.sh`

Comment (robGG1997) [reply src/ChainSafe.GamingSdk.EVM.Unity/publish-to-unity-package.sh:1]: understood

**Comment by sneakzttv** [comment]
play mode failing

**Comment by robGG1997** [comment]
LGTM
image

**Review by robGG1997 — APPROVED** [review]

**Review by juans-chainsafe — APPROVED** [review]
Looks good! playmode working again as well.

WebGL working as well

**Review by RyRy79261 — APPROVED** [review]
Tested in unity code looks good

**Pull request merged**